CRM/Core/Payment.php

```diff
@@ -1241,8 +1241,11 @@ public function getCancelUrl($qfKey, $participantID = NULL) {
     }
 
     if ($this->_component == 'event') {
-      $eventID = Participant::get(FALSE)->addWhere('id', '=', $participantID)
-        ->addSelect('event_id')->execute()->single()['event_id'];
+      $eventID = NULL;
+      if ($participantID) {
+        $eventID = Participant::get(FALSE)->addWhere('id', '=', $participantID)
+          ->addSelect('event_id')->execute()->single()['event_id'];
+      }
```

Comment:
Looks good. It can be reduced to one line: 
```
$eventID = $participantID ? Participant::get(FALSE)->addWhere('id', '=', $participantID)
        ->addSelect('event_id')->execute()->single()['event_id'] : NULL;
```